### DISCUSSION TITLE
Eloquent Product Repository example

### DISCUSSION THREAD

**Discussion by jornwildenbeest** [duncanmcclean/simple-commerce#730] [Help] [answered]
Hi,

I'm trying to store Statamics data (products, customers and orders for now) in a mysql database.

Customers and orders are working now, but I'm not sure if need to make my own repository for products.

I might be missing something here but I can't find an example of a product repository to save products in the database.
Along with an example product model.

Thanks!

**Comment by duncanmcclean** [comment, marked as answer]
Yes, the built-in Eloquent Repositories will only move customers & orders in the database, not products as most folks will want to keep them as collections so they can be version controlled.

If you do want to move products to the database though, then here might be some points of reference for you:

* [`EntryProductRepository`](https://github.com/duncanmcclean/simple-commerce/blob/4.x/src/Products/EntryProductRepository.php) - the repository for products stored as entries
* [`EloquentOrderRepository`](https://github.com/duncanmcclean/simple-commerce/blob/4.x/src/Orders/EloquentOrderRepository.php) - the repository for orders stored in the database
* [`OrderModel`](https://github.com/duncanmcclean/simple-commerce/blob/4.x/src/Orders/OrderModel.php) - the Eloquent model for orders

And if you wish to edit products in the Control Panel, you'll need to make them editable via [Runway](https://github.com/duncanmcclean/runway)

**Reply by jornwildenbeest** [reply]
Thanks for the quick reply! Understood. 

Another quick question about Runway, is possible to use the collection orderable function in the cp?
So my client can order the items in a collection using the orderable function, maybe connected to a sort column in the database or something?

**Reply by duncanmcclean** [reply]
I don't believe it is currently (unless I'm forgetting about something I've built) - could you open a Discussion on the Runway repository. I can take a proper look when I have some time.

**Reply by jornwildenbeest** [reply]
Will do, thanks
